cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -923,32 +922,34 @@ public ApplicationWithPrograms deployApp(NamespaceId namespace, @Nullable String
    * @throws Exception
```

Comment:
Does it still `throws Exception`?

Reply:
May be :). It's easier to see from the IDE, hard to say from review. What I mean is try to put specific things in throws (unless it's a really long list). It makes it easier for caller. Not it does not throw IOException anymore, so it would be good to change `throw Exception` to something specific

Reply:
git it, added throw specification here